diff --git a/include/meson.build b/include/meson.build
index a4bfaa49d..6a40d9b71 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -6,7 +6,12 @@ patch = version_split[2].to_int()
 # convert to the old-style 1.x.y version scheme used up to 1.20.x for backwards compatibility
 release = 1 * 10000000 + major * 100000 + minor * 1000 + patch
 
-dri_dep = dependency('dri', required: build_glx)
+dri_dep = dependency('dri', required: false)
+if build_glx and not dri_dep.found()
+    dri_dep = dependency('xf86driproto',
+        required: true,
+        fallback: ['xorgproto', 'ext_xorgproto'])
+endif
 
 conf_data = configuration_data()
 conf_data.set('_DIX_CONFIG_H_', '1')
